werr/errors.go

```diff
@@ -0,0 +1,35 @@
+package werr
+
+import (
+	"context"
+	"fmt"
+)
+
+// WrappedError represents a standard error
+// that can also encapsulate a context.
+type WrappedError struct {
```

Comment:
I'd rename this type to `ContextError`. All errors can be wrapped. That's not the defining characteristic of this error type. What sets this error apart is that it contains a context.